### PR TITLE
fix: only set cluster_api_url when dataplane_api not nil

### PR DESCRIPTION
**Pull request by ligfx**

### CODE REVIEW

**Comment by gene-redpanda** [comment]
Is there a particular context where this happened?

File: redpanda/resources/serverlesscluster/data_serverless_cluster.go [hunk 1]
```diff
@@ -76,19 +75,13 @@ func (d *DataSourceServerlessCluster) Read(ctx context.Context, req datasource.R
 		resp.Diagnostics.AddError(fmt.Sprintf("failed to read serverless cluster %s", model.ID), err.Error())
 		return
 	}
-	clusterURL, err := utils.SplitSchemeDefPort(serverlessCluster.DataplaneApi.Url, "443")
+	// Mapping the fields from the serverless cluster to the Terraform state
+	persist, err := generateModel(serverlessCluster)
 	if err != nil {
-		resp.Diagnostics.AddError("unable to parse Cluster API URL", err.Error())
+		resp.Diagnostics.AddError("unable to generate cluster model", err.Error())
```
Comment (gene-redpanda) [inline redpanda/resources/serverlesscluster/data_serverless_cluster.go:81]:
```suggestion
		resp.Diagnostics.AddError("unable to generate serverless cluster model", err.Error())
```

Comment (ligfx) [reply redpanda/resources/serverlesscluster/data_serverless_cluster.go]: Got rid of error entirely since the URL parsing was removed

Comment (gene-redpanda) [reply redpanda/resources/serverlesscluster/data_serverless_cluster.go]: Even better!

**Comment by ligfx** [comment]
> Is there a particular context where this happened? Meaning a cluster without a dataplane url

Most often when a cluster is deleting or wasn't destroyed correctly. I'm not sure if I've ever gotten it on a Serverless cluster but figured I'd fix it everywhere just in case and to have the same pattern.

**Review by gene-redpanda — APPROVED** [review]
LGTM!

**Pull request merged**